test/helpers/utils.go

```diff
@@ -641,9 +641,34 @@ func (kub *Kubectl) HasBPFNodePort(pod string) bool {
 	return strings.Contains(lines[0], "true")
 }
 
-// GetNodeWithoutCilium returns a name of a node which does not run cilium.
-func GetNodeWithoutCilium() string {
-	return os.Getenv("NO_CILIUM_ON_NODE")
+// GetNodesWithoutCilium returns a slice of names for nodes that do not run
+// Cilium.
+func GetNodesWithoutCilium() []string {
+	if os.Getenv("NO_CILIUM_ON_NODES") == "" {
+		if os.Getenv("NO_CILIUM_ON_NODE") == "" {
+			return []string{}
+		}
+		return []string{os.Getenv("NO_CILIUM_ON_NODE")}
+	}
+	return strings.Split(os.Getenv("NO_CILIUM_ON_NODES"), ",")
```

Comment:
Earlier it was split by a space in `test/Vagrantfile`, but this is by comma. Should these be consistent?

Reply:
Ouch, yes!

I've tested locally with a comma (e.g., `NO_CILIUM_ON_NODES=k8s2,k8s3`). The reason it worked is because the Ruby code has:
```ruby
no_cilium_nodes = $NO_CILIUM_ON_NODES.split(' ')
if no_cilium_nodes.include? "k8s#{i}"
```
So `no_cilium_nodes` contains the whole string and `include?` then works equally well on strings and arrays in Ruby. Fixed to split on commas, even if both work :smile: